### PR TITLE
Connect to API

### PR DESCRIPTION
**Pull request by lukman155**
Understand the concept of state management.
Use store, actions and reducers in React.
Connect React and Redux.
Connect an existing API via Redux and use the data to display something in a UI.

### CODE REVIEW

File: src/components/Books.js [hunk 1]
```diff
@@ -1,21 +1,31 @@
...
     <>
       <NavBar />
       <div className="books-con">
-        {books.map((book) => (
+        {(books).map((book) => (
           <Book
             key={book.id}
             title={book.title}
             author={book.author}
-            category={book.category}
```
Comment (Gambit142) [inline src/components/Books.js]: - [ ] When I reload the page, I get an error message which says id is required but it is undefined. The way you are accessing it here is wrong. It should be `book.item_id` as you defined in your reducers. Kindly implement this change. 👍

File: src/components/Form.js [hunk 2]
```diff
@@ -20,10 +20,13 @@ const Form = () => {
...
       ...book,
+      category: 'uncategorized',
       id: Date.now().toString(),
     };
-    if (book.title.length > 0 && book.title.length > 0) {
-      dispatch(addBookAction(data));
+    if (book.title.length > 0 && book.author.length > 0) {
+      dispatch(addBookAction(data)).then(() => {
+        dispatch(addBook(data));
+      });
```
Comment (Gambit142) [inline src/components/Form.js]: - [ ] When I add a book, it is not persisted through the API endpoint. I get this error in the console:

image

Although the book is persisted in the redux store, it is not persisted through the endpoint, so once I reload the page, the data is gone. Kindly fix it 👍

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @lukman155 👋, 

Good job so far!

## Highlights
- All linters error passed ✔️ 
- Descriptive pull request title ✔️ 
- Descriptive pull request summary ✔️ 

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please, remember to tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/components/Form.js (hunk 2 above)
Comment (lukman155) [reply src/components/Form.js]: @Gambit142 I have tried so many things. I have no idea why it isn't persisting in the API

Comment (Gambit142) [reply src/components/Form.js]: @lukman155, try checking the API documentation for the parameters that are needed to make a successful POST request and ensure that you attaching the right parameters. You can also check for syntax errors. 👍

**Review by Whoistolu — APPROVED** [review]
### Hi @lukman155 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**